RecoLocalFastTime/FTLClusterizer/src/MTDCPEBase.cc

```diff
@@ -0,0 +1,114 @@
+#include "Geometry/MTDGeometryBuilder/interface/MTDGeomDetUnit.h"
+#include "Geometry/MTDGeometryBuilder/interface/RectangularMTDTopology.h"
+#include "Geometry/MTDGeometryBuilder/interface/ProxyMTDTopology.h"
+
+#include "RecoLocalFastTime/FTLClusterizer/interface/MTDCPEBase.h"
+
+// MessageLogger
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+// Magnetic field
+#include "MagneticField/Engine/interface/MagneticField.h"
+
+
+#include <iostream>
+
+using namespace std;
+
+//-----------------------------------------------------------------------------
+//  A constructor run for generic and templates
+//
+//-----------------------------------------------------------------------------
+MTDCPEBase::MTDCPEBase(edm::ParameterSet const & conf,
+		       const MTDGeometry& geom)
+  : geom_(geom)
+{
+   fillDetParams();
+}
+
+
+//-----------------------------------------------------------------------------
+//  Fill all variables which are constant for an event (geometry)
+//-----------------------------------------------------------------------------
+void MTDCPEBase::fillDetParams()
```

Comment:
This is called only from the constructor, so the contents could be moved to the constructor body.